src/fosslight_util/download.py

```
@@ -1013,6 +1013,50 @@ def extract_rpm_payload(source_file: str, dest_path: str) -> bool:
 
+def _fix_extracted_permissions(path: str) -> None:
+    try:
+        os.chmod(path, os.stat(path).st_mode | 0o755)
+    except Exception:
+        pass
+    for root, dirs, files in os.walk(path, topdown=True):
+        # Fix dirs before os.walk recurses into them.
+        for d in dirs:
+            dpath = os.path.join(root, d)
+            try:
+                os.chmod(dpath, os.stat(dpath).st_mode | 0o755)
```

Ruff (0.15.15)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)